internal/download/downloadmanager.go

```diff
@@ -302,10 +302,13 @@ outer:
 					s := slabs[respIndex]
 					if s.PartialSlab {
 						// Partial slab.
-						_, err = bw.Write(s.Data)
+						n, err := bw.Write(s.Data)
```

Comment:
Make sure to also check `bw.Flush` before calling `return nil` at the end of this method to make sure we successfully flush the contents from the buffer to the response body.